### DISCUSSION TITLE
Jaxlib cuda releases only for Cuda 10.2 and cuda 11.2

### DISCUSSION THREAD

**Discussion by yashk2810** [jax-ml/jax#7698] [General]
Hello everyone,

I have been thinking about releasing GPU jaxlib wheels only for cuda 10.2 and 11.2. This makes it in line with Pytorch and TF.

What does everyone think?

**Comment by jakevdp** [comment]
I believe Colab currently uses cuda110 wheels for jaxlib, so a first order of business before making this change would be to ensure Colab is compatible with updated versions.

Edit: it appears that Colab also uses pytorch's cuda 10.2 wheels, so it may be as easy as updating the jaxlib wheels in the Colab container, and not having to touch cuda (which is a more involved change)

**Comment by yashk2810** [comment]
Is there a test I can run by installing the current 11.2 jaxlib in colab and making sure jax runs?

AFAIK, cuda is backwards compatible for 11.x versions.

**Reply by jakevdp** [reply]
We don't have any automated way to run a full test suite on Colab; the best we can do is use the GPU test notebook (https://colab.research.google.com/github/google/jax/blob/main/tests/notebooks/colab_gpu.ipynb) and add `!pip install https://storage.googleapis.com/jax-releases/cuda102/jaxlib-0.1.70+cuda102-cp37-none-manylinux2010_x86_64.whl` at the top

**Reply by skye** [reply]
I believe cuda 11.1 onwards are backwards compat, but 11.0 isn't :(

**Reply by yashk2810** [reply]
I think we'll need to ask Colab to bump their CUDA version to 11.1 at the very least. Anything above 11.1 will also be fine since 11.1 wheels work on 11.1+ versions

**Reply by jakevdp** [reply]
I just poked around the Colab container a bit... the problem there is that there is no cupy-cuda111 release that supports Python 2, and at least for the time being Python 2 seems to be part of the Colab container image (though I believe its use is deprecated). I suspect maintaining their planned deprecation schedule will be higher priority for the Colab team than having the newest version of JAX, so the effect of this change may be that jaxlib will be frozen at version 0.1.70 in Colab for the foreseeable future. I'll try a container build with cuda111 to see whether I can find a way around this.

**Reply by yashk2810** [reply]
> I'll try a container build with cuda111 to see whether I can find a way around this.

Thank you!

The freeze sounds good to me.

Or we could also build a custom wheel for colab for 11.0.

**Reply by jakevdp** [reply]
In my opinion we should do everything possible to avoid a situation where we are unable update jax in Colab – it's an important platform for JAX users & the main way we let people interact with our documentation.

**Reply by yashk2810** [reply]
Sure, we can do that (but not sustainably). Colab will need to update its Cuda version at some point.

We can build a wheel for colab only for 11.0 using the current pipeline (not the new one). The command in colab containers can be updated to point to the GCS directory right?

**Reply by yashk2810** [reply]
But I am also of the opinion that we should push colab to update to Cuda 11.1+ (maybe its in their plan?)

**Comment by yashk2810** [comment]
Now the official release for jaxlibs will be for Cuda 10.2 and Cuda 11.1 (similar to Pytorch)
